src/transformers/models/tvp/modeling_tvp.py

```diff
@@ -0,0 +1,1199 @@
+        res5_dilation = config.resnets_res_dilation
+
+        if depth not in [18, 34, 50, 101, 152]:
+            raise ValueError("The depth should be in [18, 34, 50, 101, 152]")
+        if res5_dilation not in [1, 2]:
+            raise ValueError("The res5_dilation should be in [1, 2]")
+
+        num_blocks_per_stage = {
+            18: [2, 2, 2, 2],
+            34: [3, 4, 6, 3],
+            50: [3, 4, 6, 3],
+            101: [3, 4, 23, 3],
+            152: [3, 8, 36, 3],
+        }[depth]
+
+        stages = []
+
+        # Avoid creating variables without gradients
+        # It consumes extra memory and may cause allreduce to fail
+        out_stage_idx = [{"res2": 2, "res3": 3, "res4": 4, "res5": 5}[f] for f in out_features]
+        if len(out_stage_idx) == 0:
+            raise ValueError("The element in out_features should be in [res2, res3, res4, res5]")
+        max_stage_idx = max(out_stage_idx)
+        for idx, stage_idx in enumerate(range(2, max_stage_idx + 1)):
+            dilation = res5_dilation if stage_idx == 5 else 1
+            first_stride = 1 if idx == 0 or (stage_idx == 5 and dilation == 2) else 2
+            stage_kargs = {
+                "num_blocks": num_blocks_per_stage[idx],
+                "first_stride": first_stride,
+                "in_channels": in_channels,
+                "out_channels": out_channels,
+            }
+            # Use TVPVisionBasicBlock for R18 and R34.
+            if depth in [18, 34]:
+                stage_kargs["block_class"] = TVPVisionBasicBlock
+            else:
+                stage_kargs["bottleneck_channels"] = bottleneck_channels
+                stage_kargs["dilation"] = dilation
+                stage_kargs["num_groups"] = num_groups
+                stage_kargs["block_class"] = TVPVisionBottleneckBlock
+            blocks = self.make_stage(**stage_kargs)
+            in_channels = out_channels
+            out_channels *= 2
+            bottleneck_channels *= 2
+
+            if config.backbone_freeze_at >= stage_idx:
+                for block in blocks:
+                    block.freeze()
+            stages.append(blocks)
+
+        return stages
+
+
+class TVPVisionModel(nn.Module):
+    def __init__(self, config):
+        super(TVPVisionModel, self).__init__()
+        self.in_features = config.features
+        self.backbone = TVPVisionBackbone(config)
+        self.grid_encoder = nn.Sequential(
+            nn.Conv2d(
+                config.grid_encoder_conv_input_size,
+                config.grid_encoder_conv_output_size,
+                kernel_size=3,
+                stride=1,
+                padding=1,
+                groups=1,
+                bias=False,
+            ),
+            nn.MaxPool2d(kernel_size=2, stride=2),
+            nn.ReLU(inplace=True),
+        )
+        self.input_format = config.input_format
+
+        self.config = config
+
+    def forward(self, pixel_values):
+        bsz, n_frms, c, h, w = pixel_values.shape
```

Comment:
Clearer var names - should be applied to whole method
```suggestion
        batch_size, num_frames, num_channels, height, width = pixel_values.shape
```